### DISCUSSION TITLE
Add missing apis

### DISCUSSION THREAD

**Discussion by oldma3095** [asticode/go-astiav#108] [Ideas]
example:
av_find_best_stream
av_compare_ts
swr_xxxx

**Comment by asticode** [comment]
Sure, I'll see about adding them 👍

**Reply by asticode** [reply]
FYI I've pushed both `FormatContext`.`FindBestStream` and `CompareTimestamps` on `master` 👍 

Regarding `swr_xxx`, which methods would you need? I was thinking of `swr_alloc`, `swr_free` and `swr_convert_frame` 🤔

**Reply by oldma3095** [reply]
Thanks!

swr_alloc_set_opts2
swr_init
swr_free
swr_convert

Can you provide a audio resample example with audio_fifo?

**Reply by asticode** [reply]
What are you looking to provide to `swr_convert`? 🤔  Are you working with a `*Frame` or a `[]byte`?

**Reply by oldma3095** [reply]
*Frame

**Reply by asticode** [reply]
OK then can you confirm `swr_alloc`, `swr_free` and `swr_convert_frame` would be enough instead? 🤔

**Reply by oldma3095** [reply]
> OK then can you confirm `swr_alloc`, `swr_free` and `swr_convert_frame` would be enough instead? 🤔

Just want to resample audio.I think is enough.

**Reply by asticode** [reply]
OK then, I'll check it out 👍

**Reply by asticode** [reply]
FYI I've added `SoftwareResampleContext` in `master`, you can check out the [resampling audio example](https://github.com/asticode/go-astiav/blob/master/examples/resampling_audio/main.go) to see how to use it as well as the `AudioFIFO` 👍 

I'll close this discussion but feel free to create a new issue if you're having problem using them 👍
